src/alchemlyb/visualisation/ti_dhdl.py

```diff
@@ -0,0 +1,167 @@
+"""Functions for Plotting the dhdl for the TI estimator.
+
+To assess the quality of the TI estimator, the dhdl from lambda state 0
+to lambda state 1 can plotted to assess if the change in dhdl is sampled
+thoroughly.
+
+The code for producing the dhdl plot is modified based on
+: `Alchemical Analysis <https://github.com/MobleyLab/alchemical-analysis>`_.
+
+"""
+from __future__ import division
+
+import matplotlib.pyplot as plt
+from matplotlib.font_manager import FontProperties as FP
+import numpy as np
+
+def plot_ti_dhdl(dhdl_data, labels=None, colors=None, units='kcal/mol', ax=None):
+    '''Plot the dhdl of TI.
+
+    Parameters
+    ----------
+    dhdl_data : :class:`~alchemlyb.estimators.TI` or list
+        One or more :class:`~alchemlyb.estimators.TI` estimator, where the
+        dhdl value will be taken from.
+    labels : List
+        list of labels for labelling all the alchemical transformations.
+    colors : List
+        list of colors for plotting all the alchemical transformations.
+        Default: ['r', 'g', '#7F38EC', '#9F000F', 'b', 'y']
+    units : str
+        The unit of the estimate. Default: 'kcal/mol'
+    ax : matplotlib.axes.Axes
+        Matplotlib axes object where the plot will be drawn on. If ax=None,
+        a new axes will be generated.
+
+    Returns
+    -------
+    matplotlib.axes.Axes
+        An axes with the TI dhdl drawn.
+
+    Notes
+    -----
+    The code is taken and modified from
+    : `Alchemical Analysis <https://github.com/MobleyLab/alchemical-analysis>`_
+
+    '''
+    # Make it into a list
+    try:
+        len(dhdl_data)
+    except TypeError:
+        dhdl_data = [dhdl_data, ]
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(8, 6))
+
+    ax.spines['bottom'].set_position('zero')
+    ax.spines['top'].set_color('none')
+    ax.spines['right'].set_color('none')
+    ax.xaxis.set_ticks_position('bottom')
+    ax.yaxis.set_ticks_position('left')
+
+    for k, spine in ax.spines.items():
+        spine.set_zorder(12.2)
+
+    # Make level names
+    if labels is None:
+        lv_names2 = []
+        for dhdl in dhdl_data:
+            # Assume that the dhdl has only one columns
+            lv_names2.append(dhdl.dhdl.columns.values[0].capitalize())
+    else:
+        assert len(labels) == len(dhdl_data), \
```

Comment:
Change this `assert` to a `raise ValueError`.

We should not be using assertions to capture user errors.

Reply:
If this is not tested, use `#pragma: no cover`